terminal.js

```diff
@@ -121,16 +129,23 @@ class Terminal {
         console.log(currrentPath);
     }
 
-    ls(path = FileSystemAliases.CurrentDirectory) {
+    ls(param = FileSystemAliases.CurrentDirectory) {
+        const options = { recursive: param === '-r' };
+        const path = options.recursive ? FileSystemAliases.CurrentDirectory : param;
         const root = this._getRootByPath(path);
-        let files = [];
+
+        let list = {};
         try {
-            files = this.fileSystem.readDir(path, root);
+            list = this.fileSystem.readDir(path, root, options);
         } catch(error) {
             console.log(error.message);
```

Comment:
in case of error, maybe a correct behaviour is to simply throw an the and avoid iterate an incorrect `list` object